lightning/src/routing/router.rs

```diff
@@ -1132,7 +1134,7 @@ where L::Target: Logger {
 				let features = if let Some(node_info) = $node.announcement_info.as_ref() {
 					&node_info.features
 				} else {
-					&empty_node_features
+					&variable_length_onion_optional_features
```

Comment:
I just want to clarify that I couldn't manage to create a test scenario where changing this specific line created a different result in terms of which payload format we create. Though I think it makes sense to change this line.